### ISSUE TITLE
TOTP codes generated from Aegis backup are incorrect

### ISSUE THREAD

**Issue by alanorth** [paolostivanin/OTPClient#178]
I imported a plain, unencrypted JSON Aegis backup into OTPClient, but it seems the codes generated are different than those generated by Aegis itself. I've had two sites now ask for 2FA and reject the code generated by OTPClient, only to go find my phone and use the code from Aegis successfully.

I'm using OTPClient version 2.3.0 from Flatpak. Thanks!

**Comment by paolostivanin** [comment]
If installed from the distro's repo, then it works. So it must be something related to flatpak... :thinking: 
I'll look into it

**Comment by alanorth** [comment]
I just updated the [Arch Linux AUR version to v2.3.1](https://aur.archlinux.org/packages/otpclient/#news) and I confirm that the codes imported from Aegis are correct now. I have uninstalled the flatpak version.

**Comment by paolostivanin** [comment]
Yep, I reproduced the bug yesterday. It happens only on Flatpak for whatever reason :thinking:

**Comment by paolostivanin** [comment]
Weird. I was able to reproduce it using flatpak-builder on my dev machine, but on a newly installed system the flatpak version works just fine :astonished:

**Comment by alanorth** [comment]
OK so you have some debugging to do. Bummer! I will stick to the non-Flatpak version from AUR now that I got it updated to v2.3.1 (and also I'll ask the derelict maintainer to assign the package to me). You can close this issue if you like. Cheers!
